dashboard.py

```diff
@@ -17,51 +17,73 @@
 panel.extension("vega")
 
 
-def get_test_source():
-    """
-    Crawl the tests directory and try to grab code for each test on a best-effort
-    basis. This relies on the tests being importable from this script, so the
-    environment should be similar enough that that is possible.
-    """
-    source: dict[str, str] = {}
-    files = glob.glob("tests/**/test_*.py", recursive=True)
-    for f in files:
-        try:
-            # Fragile!
-            mod = importlib.import_module(f.replace("/", ".")[: -len(".py")])
-            tests = [a for a in dir(mod) if a.startswith("test_")]
-            for test in tests:
-                if fn := getattr(mod, test, None):
-                    if not callable(fn):
-                        continue
-                    source[f[len("tests/") :] + "::" + test] = inspect.getsource(fn)
-        except BaseException:  # Some pytest exceptions inherit directly from BaseException
-            pass
-    return source
-
-
-source = get_test_source()
-
-
-def make_timeseries(originalname, df, spec) -> altair.Chart | None:
+source: dict[str, str] = {}
+
+
+def load_test_source() -> None:
+    """Crawl the tests directory and try to grab code for each test. This relies on the
+    tests being importable from this script.
     """
-    Make a single timeseries altair chart for a given test.
+    for fname in glob.iglob("tests/**/test_*.py", recursive=True):
+        mod = importlib.import_module(fname.replace("/", ".")[: -len(".py")])
+        tests = [a for a in dir(mod) if a.startswith("test_")]
+        for test in tests:
+            if (func := getattr(mod, test, None)) and callable(func):
+                # FIXME missing decorators, namely @pytest.mark.parametrize
+                source[fname[len("tests/") :] + "::" + test] = inspect.getsource(func)
 
-    originalname: str
-        The name of the test without any fixture or other modifications.
 
-    df: pandas.DataFrame
-        A dataframe with the test data in it.
+def make_barchart(
+    df: pandas.DataFrame, spec: altair.ChartSpec, title: str
+) -> altair.Chart | None:
+    """Make a single Altair barchart for a given test."""
+    df = df.dropna(subset=[spec.field, "start"])
+    if not len(df):
+        # Some tests do not have average_memory or peak_memory measures, only runtime
+        return None
 
-    spec: ChartSpec
-        Data for how to render the timeseries
-    """
+    df = df.sort_values("runtime", key=runtime_sort_key_pd)
+    df = df.assign(**{spec.field: df[spec.field] / spec.scale})
+    df = df[
+        [
+            spec.field,
+            "name",
+            "dask_version",
+            "distributed_version",
+            "runtime",
+        ]
+    ]
+    # Altair will re-sort alphabetically, unless it's explicitly asked to sort by
+    # something else. Could not find an option to NOT sort.
+    df["order"] = pandas.RangeIndex(0, df.shape[0])
```

Comment:
@ian-r-rose any ideas?

Reply:
You may be able to do this with explicitly passing `sort=None`